mavros/include/mavros/frame_tf.h

```diff
@@ -386,12 +386,12 @@ inline void covariance_to_mavlink(const T &cov, std::array<float, SIZE> &covmsg)
 }
 
 /**
- * @brief Store half upper right triangular of 9d Covariance matrix to MAVLink float[n] format
+ * @brief Convert half upper right triangular of a covariance matrix to MAVLink float[n] format
  */
-inline void covariance9d_urt_to_mavlink(const Covariance9d &cov, std::array<float, 45> &covmsg)
+template<class T, std::size_t SIZE>
+inline void covariance_urt_to_mavlink(const T &covmap, std::array<float, SIZE> &covmsg)
```

Comment:
Now it should have assert that number of URT elements of covmap == SIZE.

Also it may be a good idea to keep `covariance9d_...()` function (just change to call new one).

Reply:
> Now it should have assert that number of URT elements of covmap == SIZE.

Added now.

> Also it may be a good idea to keep `covariance9d_...()` function (just change to call new one).

For what reason, given that this generalizes the implementation?

Reply:
Use `ROS_ASSERT()`.

Reply:
> Use `ROS_ASSERT()`.

Makes sense. Done.